src/pow.rs

```diff
@@ -0,0 +1,1889 @@
+// Rust Bitcoin Library - Written by the rust-bitcoin developers.
+// SPDX-License-Identifier: CC0-1.0
+
+//! Proof-of-work related integer types.
+//!
+//! Provides the [`Work`] and [`Target`] types that are use in proof-of-work calculations. The
+//! functions here are designed to be fast.
```

Comment:
I'm not sure if I've broken this statement now with everything I've done?

Reply:
I think they are still fast but I'm not sure if that's a very important property.

Reply:
Agreed on both counts. I think "designed to be fast" is supposed to mean "will not cause a DoS vector if you are syncing a node and using these functions to vet potential header chains" ... but like, our arithmetic would have to be *really* slow for that to be your biggest problem.

Reply:
Saying it's designed to be safe to check headers would be fine, I think.